docs/async/async-fsharp.rst

```diff
@@ -0,0 +1,186 @@
+Async Programming in F#
+=======================
+By `Phillip Carter`_
+
+Async programming in F# can be accomplished through a language-level programming model designed to be easy to use and look just like any other F# code.
```

Comment:
It looks like some of my feedback from the C# doc can be applied here. I won't go line by line until you've had an opportunity to do that.